test/planned_subvol_test.rb

```diff
@@ -207,4 +208,122 @@
     end
   end
 
+  context ".read_from_xml" do
```

Comment:
On the other hand, this looks more like a set of examples than a typical RSpec test to me (more on that later).